src/@batch-flask/utils/secure-utils.ts

```diff
@@ -16,17 +16,23 @@ export class SecureUtils {
     public static username(length: number = 6): string {
         const charset = "abcdefghijklmnopqrstuvwxyz";
         let retVal = "";
-        for (let i = 0, n = charset.length; i < length; ++i) {
-            retVal += charset.charAt(Math.floor(Math.random() * n));
-        }
+        // create an array and fill with random integer
+        const randomArray = new Uint32Array(length);
+        window.crypto.getRandomValues(randomArray);
+        for (let i = 0; i < length; i++) {
+            retVal += charset.charAt(randomArray[i] % charset.length);
+        }        
         return retVal;
     }
 
     public static password(length: number = 15): string {
         const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+~`|}{[]\:;?><,./-=";
         let retVal = "";
-        for (let i = 0, n = charset.length; i < length; ++i) {
-            retVal += charset.charAt(Math.floor(Math.random() * n));
+        // create an array and fill with random integer
+        const randomArray = new Uint32Array(length);
+        window.crypto.getRandomValues(randomArray);
+        for (let i = 0; i < length; i++) {
+            retVal += charset.charAt(randomArray[i] % charset.length);
```

Comment:
can you use something else than appending to a string here for performance(Maybe a buffer or just an array)